js/src/LCSDiff.jsx

```diff
@@ -68,14 +81,31 @@ export default class LCSDiff extends React.Component {
     const {
       width,
       height,
+      progress,
     } = this.state;
 
     return (
-      <canvas
-        ref={(node) => { this.canvas = node; }}
-        width={width}
-        height={height}
-      />
+      <div>
+        {progress === 0 &&
+          <ReactWaypoint
+            onEnter={this.initialize}
+            scrollableAncestor={window}
```

Comment:
Should we add an offset so we can start loading things before they scroll into the viewport? I think that would make the interface feel better.